File: 1380-lucky-numbers-in-a-matrix/1380-lucky-numbers-in-a-matrix.py
class Solution:
    def luckyNumbers (self, matrix: List[List[int]]) -> List[int]:
        m, n = len(matrix), len(matrix[0])
        rowMin = [0] * m
        ans = []

        for i in range(m):
            mn = matrix[i][0]
            for j in range(1, n):
                mn = min(mn, matrix[i][j])
            rowMin[i] = mn

        for j in range(n):
            mx = matrix[0][j]
            for i in range(1, m):
                mx = max(mx, matrix[i][j])
            
            for i in range(m):
                if rowMin[i] == mx:
                    ans.append(mx)
                    break

        return ans

File: 1380-lucky-numbers-in-a-matrix/1380-lucky-numbers-in-a-matrix.java
class Solution {
    public List<Integer> luckyNumbers (int[][] matrix) {
        int m = matrix.length, n = matrix[0].length;
        var rowMin = new int[m];
        var ans = new ArrayList<Integer>();

        for(var i=0; i<m; i++){
            var mn = matrix[i][0];
            for(var j=1; j<n; j++)
                mn = Math.min(mn, matrix[i][j]);
            rowMin[i] = mn;
        }

        for(var j=0; j<n; j++){
            var mx = matrix[0][j];
            for(var i=1; i<m; i++)
                mx = Math.max(mx, matrix[i][j]);
            
            for(var i=0; i<m; i++)
                if(rowMin[i] == mx){
                    ans.add(mx);
                    break;
                }
        }

        return ans;
    }
}